src/wp-includes/default-constants.php

```diff
@@ -97,6 +97,15 @@ function wp_initial_constants() {
 		define( 'WP_DEBUG_LOG', false );
 	}
 
+	/*
+	 * Add define( 'WP_DEVELOPMENT_MODE', 'core' ) or define( 'WP_DEVELOPMENT_MODE', 'plugin' ) or
+	 * define( 'WP_DEVELOPMENT_MODE', 'theme' ) to wp-config.php to signify development mode for WordPress core, a
+	 * plugin, or a theme respectively.
+	 */
+	if ( ! defined( 'WP_DEVELOPMENT_MODE' ) ) {
+		define( 'WP_DEVELOPMENT_MODE', defined( 'WP_RUN_CORE_TESTS' ) ? 'core' : '' );
+	}
```

Comment:
Don't think it is a good idea to force to run the core tests in development mode. The opposite, core tests should be able to run in both dev. mode and production mode. Perhaps another ticket/PR for that may be needed if it doesn't exist already.

Reply:
@azaozz Fair point, I've updated this to just set to development mode disabled (i.e. `''`) by default. Let's open a separate ticket for considering whether to run tests in multiple development modes.